privval/messages.go

```diff
@@ -0,0 +1,61 @@
+package privval
+
+import (
+	amino "github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/types"
+)
+
+// RemoteSignerMsg is sent between SignerServiceEndpoint and the SignerServiceEndpoint client.
```

Comment:
Does it make sense to rename

SignerServiceEndpoint -> SignerRemoteServer
SignerServiceEndpoint -> SignerRemoteClient

?

Reply:
I intentionally tried to avoid client/server here, because the concept it is actually mixed up in this implementation.

- Tendermint opens a listening socket  (server like)
- KMS dials in (client like)
- Tendermint initiates requests over the connection (client like)
- KMS answers (server like)

I tried to avoid this confusion in the names.
In the next PR, I will encapsulate this behind the Signer, which at the moment is the opposite too.

Reply:
Makes sense. Why the endpoint suffix then? And not just `SignerService`?

Reply:
I think both `SignerServiceEndpoint` or simply `SignerService` are better than `SocketVal`. Endpoint makes clear it is a connection endpoint. But yeah, let's get this right this time. The renaming is the main purpose of this PR.

Reply:
My intention is to separate and encapsulate the "network related" code from the actual signer.

SignerRemote ---(uses/contains)--> SignerValidatorEndpoint <--(connects to)--> SignerServiceEndpoint --->  SignerService (future.. not here yet but would like to decouple too)

All reconnection/heartbeat/whatever code goes in the endpoints. Signer[Remote/Service] do not need to know about that.

I agree Endpoint may not be the perfect name. I tried to find something "Go-ish" enough. It is a common name in go-kit, kubernetes, etc. 

I am happy to change it as long as it makes clear the difference between network and non-network related.

What do you think?

Reply:
I think your reasoning behind this makes a lot of sense. Maybe we can add this:

> SignerRemote ---(uses/contains)--> SignerValidatorEndpoint <--(connects to)--> SignerServiceEndpoint ---> SignerService (future.. not here yet but would like to decouple too)

in a doc.go package documentation.

Reply:
Yes, but it is actually not working this way yet..  
that is why we agreed on rename a few things before changing/breaking things :)

SignerValidatorEndpoint: 
- handles the listener
- contains SignerRemote
- Implements the PrivValidator interface
- connects and sets a connection object in a contained SignerRemote
- delegates PrivValidator some calls to SignerRemote which in turn uses the conn object that was set externally

SignerRemote:
- Implements the PrivValidator interface
- handles heartbeats

SignerServiceEndpoint: 
- Does most things in a single place 
- delegates to a PrivValidator IIRC.

Reply:
Alright, then let's not document this now but when the refactoring is finished :-)

Reply:
Thanks for explaining this 🙇 LGTM